docs/user-guides/scripts.md

```diff
@@ -0,0 +1,423 @@
+# Scripts
+
+Scripts are an essential part of Hera's extension on top of Argo. As Hera is a Python library, Python Script templates
+become the standard template, which is reflected by the greater feature set provided for writing them.
+
+## Script Decorator
+
+The `script` decorator function is a key offering of Hera to achieve near-native Python function orchestration. It
+allows you to call the function under a Hera context manager such as a `Workflow` or `Steps` context, and it will be
+treated as the intended sub-object, which would be a `template` when under a `Workflow`, or a `Step` when under a
+`Steps`. The function will still behave as normal outside of any Hera contexts, meaning you can write unit tests on the
+given function.
+
+> **For advanced users**: the exact mechanism of the `script` decorator is to prepare a `Script` object within the
+> decoration, so that when your function is invoked under a Hera context, the call is redirected to the
```

Comment:
```suggestion
> decorator, so that when your function is invoked under a Hera context, the call is redirected to the
```

Reply:
Yeah wasn't sure which word to use there 🤣 "decorator" sounds fine